core/dynamo.c

```diff
@@ -979,6 +984,9 @@ dynamo_shared_exit(thread_record_t *toexit /* must ==cur thread for Linux */
      */
     instrument_exit();
 #endif
+    synch_with_threads_at_exit(exit_synch_state(), false/*post-exit*/);
```

Comment:
Should this be inside the ifdef as it will be a waste of time to grab the lock and check for threads when it's impossible for there to be any threads left, right?